### PR TITLE
Add methods to toggle disable/enable sharing settings

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds toggle functions to disabling/enabling sharing settings.

## Related Issue
#31983

## How Has This Been Tested?
Locally

## Types of changes
- [x] Tests

## Checklist:
- [x] Acceptance tests added

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 1]
```diff
@@ -64,132 +64,154 @@ public function theAdminBrowsesToTheAdminSharingSettingsPage() {
...
-	 * @When the admin disables the share API using the webUI
+	 * @When the admin changes the state of share API to :requiredState using the webUI
+	 * @param string $requiredState
 	 *
 	 * @return void
 	 */
-	public function theAdminDisablesTheShareApiUsingTheWebUI() {
-		$this->adminSharingSettingsPage->disableShareApi();
+	public function adminTogglesShareApiUsingTheWebui($requiredState) {
+		$this->getSession()->reload();
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: Why do we need the reloads everywhere?

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 2]
```diff
@@ -64,132 +64,154 @@ public function theAdminBrowsesToTheAdminSharingSettingsPage() {
...
+		$this->adminSharingSettingsPage->toggleShareApi($requiredState);
 		$this->adminSharingSettingsPage->waitForAjaxCallsToStartAndFinish(
 			$this->getSession()
 		);
 	}
 
 	/**
-	 * @When the admin disables share via link using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: You could use regular expression in the step to have the same line for enable/disable, then you could use the old steps and don't need to change existing scenarios

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 3]
```diff
@@ -36,141 +36,293 @@ class AdminSharingSettingsPage extends OwncloudPage {
...
 	/**
-	 * Disable the Share API
+	 * toggle the Share API
+	 *
+	 * @param string $requiredState
 	 *
 	 * @return void
 	 */
-	public function disableShareApi() {
+	public function toggleShareApi($requiredState) {
```
Comment (individual-it) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: A lot of duplicated in the toggle functions, maybe we can combine them?

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 4]
```diff
@@ -36,141 +36,218 @@ class AdminSharingSettingsPage extends OwncloudPage {
...
+	 * @param string $checkboxId
 	 *
 	 * @return void
 	 */
-	public function disableShareApi() {
-		$shareApiCheckbox = $this->find("xpath", $this->shareApiCheckboxXpath);
-		$shareApiCheckbox->click();
+	public function toggleCheckbox($action, $checkboxXpath, $checkboxId) {
+		$checkbox = $this->find("xpath", $checkboxXpath);
+		$checkCheckbox = $this->findById($checkboxId);
```
Comment (individual-it) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: please check if `find*()` returns `null`

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 5]
```diff
@@ -36,141 +36,218 @@ class AdminSharingSettingsPage extends OwncloudPage {
...
+				__METHOD__ . " invalid action: " . $action
+			);
+		}
 	}
 
 	/**
-	 * Disable users to share via link
+	 * toggle the Share API
+	 *
+	 * @param string $action e.g "|enables|disables|"
```
Comment (individual-it) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: not "e.g." only "enables" or "disables" can be used

File: tests/acceptance/features/webUIAdminSettings/adminSharingSettings.feature [hunk 6]
```diff
@@ -4,82 +4,189 @@ Feature: admin sharing settings
...
+	Scenario: disable enforce password protection for upload-only links
+		Given parameter "shareapi_enforce_links_password_write_only" of app "core" has been set to "no"
+		And the admin has browsed to the admin sharing settings page
+		When the admin disables enforce password protection for upload only links using the webUI
+		And the user retrieves the capabilities using the API
+		Then the capabilities should contain
+			| capability    | path_to_element                                | value |
+			| files_sharing | public@@@password@@@enforced_for@@@upload_only | EMPTY |
+
+		Scenario: enable resharing
```
Comment (individual-it) [inline tests/acceptance/features/webUIAdminSettings/adminSharingSettings.feature]: wrong indent

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by paurakhsharma** [comment]
@individual-it can you please review it again. Thank you :slightly_smiling_face:

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
looks good lets see what :robot: says
please backport

**Comment by paurakhsharma** [comment]
@PVince81 Some problem with codecov. Can you please merge this PR.

**Comment by PVince81** [comment]
please backport

**Comment by paurakhsharma** [comment]
Ok, thank you

**Comment by paurakhsharma** [comment]
Backport on #32054

**Pull request merged**